src/Integration/NewConnectedMode/IConfigurationProvider.cs

```diff
@@ -0,0 +1,35 @@
+/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarLint.VisualStudio.Integration.Persistence;
+
+namespace SonarLint.VisualStudio.Integration.NewConnectedMode
+{
+    internal interface IConfigurationProvider : ILocalService
+    {
+        /// <summary>
+        /// Returns the currently bound project, or null if the current solution is
+        /// not bound, or if there is not a current solution
+        /// </summary>
+        BoundSonarQubeProject GetBoundProject();
+
+//        ConnectedMode GetMode();
```

Comment:
I'm undecided whether to add _GetMode()_ to this interface, or to add a _BindingMode_ property to _BoundSonarQubeProject_. I'll see how it goes, but feel free to make suggestions.
Having a _BoundSonarQubeProject.BindingMode_ property that could be _Standalone_ seems a bit weird.